### PR TITLE
test: Add more tests for updating of package.json files

### PR DESCRIPTION
**Pull request by msrose**
These tests still hard code the `@thm` scope, but we basically want to make sure we preserve functionality as we rip out this Top Hat-specific logic, so the first step is writing tests.

### CODE REVIEW

File: test/deployPackages.test.js [hunk 1]
```diff
@@ -60,6 +62,35 @@ describe('deployPackages function', () => {
         })
     })
 
+    const getPackageJsonContents = packageName =>
+        fs.readFileSync(
+            path.join(
+                process.cwd(),
+                'packages',
+                packageName.replace('@thm/', ''),
```
Comment (msrose) [inline test/deployPackages.test.js]: This will change once we remove the Top Hat-specific logic from the code

File: test/setup.js [hunk 2]
```diff
@@ -8,29 +8,58 @@ jest.mock('../src/command-helpers')
...
                 }),
             {},
         ),
-        packagesRoot,
+        publishConfig: {},
+    }
+}
+
+function getPackageJsonPath(packageName) {
+    return `${packageName.replace('@thm/', '')}/package.json`
```
Comment (msrose) [inline test/setup.js]: This will also change when we generalize the Top Hat-specific stuff

File: test/setup.js [hunk 3]
```diff
@@ -8,29 +8,58 @@ jest.mock('../src/command-helpers')
...
+                    buildPackageJson(packageName, version, dependencies),
+                ),
+            }),
+        {},
     )
+}
+
+beforeEach(() => {
+    vol.reset()
+    vol.fromJSON(getMockPackageJsonFiles(mockPackages), packagesRoot)
```
Comment (iamogbz) [inline test/setup.js]: Note to self use `memfs` in `yvm`

**Comment by msrose** [comment]
@all-contributors please add @iamogbz for review

**Pull request merged**